File: python/goorm/2. 알고리즘의 연산량/B.cpp
#include<cstdio>
#include<iostream>

using namespace std;

/**
* 주어진 배열이 오름차순인지 검사하는 함수
* @param data
* @param n     데이터의 수
* @return      data[0] ~ data[n-1]이 오름차순이라면 true, else false
*/
bool isOrdered(int data[], int n)
{
	for (int i=0; i<n-1; i++) {
        if (data[i] > data[i+1]) return false;
    }
    return true;
}

int main()
{
	int n;
	int* data;

	scanf("%d", &n);
	data = new int[n];

	for (int i = 0; i < n; i++)
	{
		scanf("%d", &data[i]);
	}

	bool result = isOrdered(data, n);

	if (result)
	{
		printf("YES");
	}
	else{
		printf("NO");
	}

	delete[] data;
	return 0;
}

File: python/goorm/2. 알고리즘의 연산량/B.py



def isOrdered(data, n):
    for i in range(n - 1):
        if data[i] > data[i+1]:
            return False
    return True

if __name__ == "__main__":
    n = int(input())
    data = list(map(int, input().split()))

    result = isOrdered(data, n)

    print("YES" if result == True else "NO")